hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveMetastore.java

```diff
@@ -144,13 +171,35 @@ public void stop() {
     if (executorService != null) {
       executorService.shutdown();
     }
-    if (hiveLocalDir != null) {
-      hiveLocalDir.delete();
-    }
     if (baseHandler != null) {
       baseHandler.shutdown();
     }
     METASTORE_THREADS_SHUTDOWN.invoke();
+    HMS_HANDLER_THREAD_LOCAL_CONF.remove();
+    HMS_HANDLER_THREAD_LOCAL_TXN.remove();
+    Object connPool = CONN_POOL.get();
+    CONN_POOL.set(null);
+    if (connPool instanceof Closeable) {
+      try {
+        ((Closeable) connPool).close();
+      } catch (IOException e) {
+        throw new RuntimeException("Failed to close connection pool in TxnHandler", e);
+      }
+    }
+    OBJECT_STORE_PROP.set(null);
+    PersistenceManagerFactory pmf = OBJECT_STORE_PMF.get();
```

Comment:
These should be cleaned up immediately when there is a configuration change in the connection pool URI.
Maybe our tests are not setting the connection pool URI correctly?

Reply:
Yeah it seems `OBJECT_STORE_PROP` and `OBJECT_STORE_PMF` will be cleaned up when connection URI changes. I took a conservative approach to clean them anyway because I thought it does no harm and is safer than relying on hive implementation details (especially when we have multiple versions to support). If you prefer not to do it, I'll revert the change.